api/assessment/assessment.go

```diff
@@ -25,22 +25,31 @@
 
 package assessment
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+)
 
 type ResultHookFunc func(result *AssessmentResult, err error)
 
 var (
+	ErrAssessmentIdConfiguredIncorrectly     = errors.New("assessment result id is configured incorrectly")
```

Comment:
Maybe `ErrIdInvalidFormat` ? Configured correctly sounds misleading. It should say that the Id is not in an expected format (UUID)